components/builder-api-client/src/builder.rs

```diff
@@ -462,6 +462,33 @@ impl BuilderAPIProvider for BuilderAPIClient {
                       progress)
     }
 
+    /// Create an origin
+    ///
+    ///  # Failures
+    ///
+    ///  * Remote builder is not available
+    ///  * Origin already exists
+    ///  * Y'aint got no auth
+    fn create_origin(&self, origin: &str, token: &str) -> Result<()> {
+        let body = json!({
+            "name": origin,
+        });
+
+        let sbody = serde_json::to_string(&body)?;
+
+        let res = self.add_authz(self.0.post("depot/origins"), token)
+                      .body(&sbody)
```

Comment:
You can eliminate an additional binding* and avoid having to handle an error that can't really occur by using `serde_json::Value`'s `to_string` method:
```suggestion
                      .body(&body.to_string())
```

\* Not that additional bindings are bad, but the fewer things to keep in one's head, the easier I find it to reason about code. Another alternative is to re-bind `body` as a `String` rather than introduce `body`. Shadowing may be something to avoid in other languages, but it's [idiomatic rust](https://doc.rust-lang.org/book/ch03-01-variables-and-mutability.html?highlight=shadow#shadowing).

Reply:
This and a couple other serde handings will get fixed in my PR with reqwest